File: Lib/Rule.cpp
#include <Rule.h>
Rule::Rule(const unsigned short ruleNumber)
{
	for (unsigned short i = 0, unsigned short divider = 128; i<8; ++i, divider /= 2)
	{
		state[i] = ruleNumber%divider;
	}
}
unsigned short Rule::getState(const unsigned short neighborhood) const
{
	return state[neighborhood];
}

File: Lib/ca/Rule.java
package ca;

public class Rule {
    private boolean[] state;

    public Rule(int number) {
        state = new  boolean[8];
        for (int i = 0; i < 8; ++i) {
            state[i] = (number & (1 << i)) != 0;
        }
    }

    public boolean getAt(int neighbourhood) {
        return state[neighbourhood];
    }
}
